pyrdfstore/store.py

```diff
@@ -0,0 +1,276 @@
+        Note: dropping any unknown graph should just work without complaints
+        Note: dropping a graph still leaves a trail of its 'update'
+              in the admin-graph (meaning its age can be verified)
+              Consider forget_graph to remove that trail of 'update'
+
+        :param named_graph: the uri describing the named_graph to drop
+        :type named_graph: str
+        :rtype: None
+        """
+        pass
+
+    @abstractmethod
+    def forget_graph(self, named_graph: str) -> None:
+        """forgets about the names_graph being under control
+        This functions independent of the drop_graph method.
+        So any client of this service is expected to decide when (or not) to combine both
+
+        Note: dropping any unknown graph should just work without complaints
+        Note: forgetting a graph removes any trail of its 'update'
+              in the admin-graph.
+              It does nothing to also actually removed the content in the graph
+
+        :param named_graph: the uri describing the named_graph to drop
+        :type named_graph: str
+        :rtype: None
+        """
+        pass
+
+
+class URIRDFStore(RDFStore):
+    """This class is used to connect to a SPARQL endpoint and execute SPARQL queries
+
+    :param read_uri: The URI of the SPARQL endpoint to read from
+    :type read_uri: str
+    :param write_uri: The URI of the SPARQL endpoint to write to.
+                        If not provided, the store can only be read from, not updated.
+    :type write_uri: Optional[str]
+    """
+
+    def __init__(self, read_uri: str, write_uri: Optional[str] = None):
+        self.sparql_store = None
+        self.allows_update = False
+        if write_uri is None:
+            self.sparql_store = SPARQLStore(query_endpoint=read_uri)
+        else:
+            self.sparql_store = SPARQLUpdateStore(
+                query_endpoint=read_uri,
+                update_endpoint=write_uri,
+                method="POST",
+                autocommit=True,
+            )
+            self.allows_update = True
+
+    def select(self, sparql: str, named_graph: Optional[str] = None) -> Result:
+        log.debug(f"exec select {sparql=} into {named_graph=}")
+        if named_graph is not None:
+            select_graph = Graph(
+                store=self.sparql_store, identifier=named_graph
+            )
+        else:
+            select_graph = Graph(store=self.sparql_store)
+        result: Result = select_graph.query(sparql)
+        log.debug(f"from SPARQLStore :: {type(result)=} -> {result=}")
+        return result
+
+    def insert(self, graph: Graph, named_graph: Optional[str] = NIL_NS):
+        assert (
+            self.allows_update
+        ), "data can not be inserted into a store if no write_uri is provided"
+        log.debug(f"insertion of {len(graph)=} into ({named_graph=})")
+        store_graph = Graph(store=self.sparql_store, identifier=named_graph)
+        try:
```

Comment:
are we sure we want this ? 
this hides runtime failures as it demotes them to mere log-file-entries